### ISSUE TITLE
Incorrect `typecheck` failure

### ISSUE THREAD

**Issue by jidicula** [golangci/golangci-lint-action#408]
I'm seeing an incorrect failure of `typecheck` invoked by `golangci-lint-action@v3.1.0` in GitHub Actions: https://github.com/jidicula/vci-check/runs/5345676647?check_suite_focus=true#step:3:24

Specifically, it's throwing the error: `Error: ReadAll not declared by package io (typecheck)`

This is clearly incorrect, as `io.ReadAll()` indeed exists: https://pkg.go.dev/io#ReadAll . Perhaps it's parsing it incorrectly as a type exported by `io` rather than an exported function?

Running `golangci-lint` 1.44.2 locally on my machine does not throw this error.

**Comment by SVilgelm** [comment]
Please read the [README](https://github.com/golangci/golangci-lint-action#compatibility, the v3 does not install go anymore, you need to add:
```yaml
- uses: actions/setup-go@v2
        with:
          go-version: '1.17'
```

**Comment by bewuethr** [comment]
I see incorrect typecheck failures as well, but I do set up Go:

```yaml
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3.0.0
        with:
          token: ${{ secrets.PAT_TOKEN }}

      - name: Setup Go
        uses: actions/setup-go@v3.0.0
        with:
          go-version: 1.17.7

      - name: Run golangci-lint
        # yamllint disable-line rule:line-length
        uses: golangci/golangci-lint-action@b517f99ae23d86ecc4c0dec08dcf48d2336abc29  # v3.1.0
        env:
          GOPRIVATE: github.com/myorg
        with:
          version: v1.44.2
```

**Comment by GiaoGiaoCat** [comment]
Here is my config and it fixed the problem.

```
name: golangci-lint
on:
  push:
    tags:
      - v*
    branches:
      - main
  pull_request:
  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

jobs:
  golangci:
    name: lint
    runs-on: ubuntu-latest
    steps:
      - name: Setup Go
        uses: actions/setup-go@v3.0.0
        with:
          go-version: 1.17.8
          
      - uses: actions/checkout@v2
      
      - name: Check for unused Golang imports
        uses: Jerome1337/gomod-action@v1.0.0
        
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v3.1.0
        with:
          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
          version: v1.45.2

          # Optional: working directory, useful for monorepos
          working-directory: ./be

          # Optional: golangci-lint command line arguments.
          # args: --issues-exit-code=0
          args: -v

          # Optional: show only new issues if it's a pull request. The default value is `false`.
          # only-new-issues: true

          # Optional: if set to true then the action will use pre-installed Go.
          # skip-go-installation: true

          # Optional: if set to true then the action don't cache or restore ~/go/pkg.
          # skip-pkg-cache: true

          # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
          # skip-build-cache: true
```

**Comment by lonnblad** [comment]
I have this problem currently as well with this config:
```
  golangci:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/setup-go@v3
        with:
          go-version: "1.19.2"
      - uses: actions/checkout@v3
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v3.2.0
        with:
          version: v1.50
```

**Comment by lonnblad** [comment]
My error was related to that a dependency to a private module didn't get downloaded correctly.
Really weird to get a typecheck-lint-error on this though.
